nsqd/backend_queue.go

```diff
@@ -9,4 +11,5 @@ type BackendQueue interface {
 	Delete() error
 	Depth() int64
 	Empty() error
+	WriteMsg(buf *bytes.Buffer, msg *Message) error
```

Comment:
It was intentional for `DiskQueue` to not have first class knowledge of `Message` - that's why we had the shim in between the two.